src/main/common/time.c

```diff
@@ -238,6 +238,17 @@ bool rtcGet(rtcTime_t *t)
     return true;
 }
 
+bool rtcGetWithOffset(rtcTime_t *t, int16_t minutes)
+{
+    rtcTime_t workTime;
+    if (rtcGet(&workTime)) {
+        *t = workTime + (minutes * 60 * MILLIS_PER_SECOND);
```

Comment:
This should check for underflow.

Reply:
The `rtcTime_t` type is `int64_t` so underflow really isn't a concern.

Reply:
Interesting. I had thought it started at the beginning of the epoch.

Reply:
It is, but there were dates before 1970 ;). So the standard implementation (and how the RTC code is built) uses signed values.  I agree that in our context it really doesn't make sense. But keeping our library per the standard is probably a good idea in case we interact with other systems that speak "unix epoch time".

Reply:
Agreed there. That was also the motivation for my suggestion to stick the full RTC timestamp into the permanent memory.

Reply:
This is such a limited use-case though that the loss of precision to seconds (or even minutes) is fine so no real reason to persist the entire millisecond-level 64bit timestamp.

When I started this I did investigate generally persisting the RTC time across reboots and resetting on initialization. But it would take hitting every possible occurrence of the reboot code and there are a lot. Without doing it deep in the STM libraries and changing `NVIC_SystemReset()` it would be a lot of work to pull off. And for probably questionable value as the RTC time would be behind anyway (due to the reboot/init process) and there are a limited number of use cases where having the RTC be restored after a reboot (assuming it was previously set) are of value. The main one being for MSC file timestamps so I went with a more vertical approach since we had a clear exit point for rebooting into MSC.

Also, dealing with actual persisted value is painful. We have to assume that there will be cases where the value won't be set (like a cold boot). So we have to presume a 0 value is unset. But a 0 value is technically valid timestamp. However we can ignore that in our limited use case as we're unlikely to have a 1970 date for the flight controller RTC and we can just consider that to be invalid. Becomes more challenging if considering a generalized RTC persistence.